x-pack/functionbeat/scripts/mage/update.go

```diff
@@ -46,3 +48,75 @@ func (Update) IncludeFields() error {
 
 	return devtools.GenerateAllInOneFieldsGo()
 }
+
+// VendorBeats collects the vendor folder required to deploy the function for GCP.
+func (Update) VendorBeats() error {
+	gcpVendorPath := filepath.Join("provider", "gcp", "build", "vendor")
+	vendorPath := filepath.Join("..", "..", "vendor")
+	beatsVendorPath := filepath.Join(gcpVendorPath, "github.com", "elastic", "beats")
+
+	cp := &devtools.CopyTask{
+		Source: vendorPath,
+		Dest:   gcpVendorPath,
+		Mode:   0600,
+		Exclude: []string{
+			".*/awslabs",
+			".*/goracle",
+			".*/k8s.io",
+			".*/docker",
+			".*/Azure",
+			".*/govmomi",
+			".*/prometheus",
+			".*/aws",
+			".*/go-mssql",
+			".*/aerospike",
+			".*/zmap",
+			".*/go-sytemd",
+			".*/godbus",
+			".*/gogo",
+			".*/howett.net",
+			".*/grpc",
+			".*/genproto",
+			".*/gopkg.in",
+			".*/appengine",
+			".*/containerd",
+			".*/go-sql-driver",
+			".*/sarama",
+			".*/klauspost",
+			".*/gopacket",
+			".*/shirou",
+			".*/4d64.com",
+			".*/go/storage",
+			".*go.opencensus.io",
+			".*/x/sys/windows",
+		}, // exclude big unused deps
+	}
```

Comment:
Quite a long list. The command `go list -deps` should allow us to create a list of transitive dependencies. Maybe create a follow up issue here?